### PR TITLE
fix: allow forwarding query parameters to the session store

### PR DESCRIPTION
**Pull request by grantzvolsky**
This commit adds a new flag, preserve_query, in order to resolve issue #786. It
would be ideal for this flag to be false by default for consistency with
preserve_path, but it is true by default in order to preserve backwards
compatibility.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Comment by aeneasr** [comment]
Let me know when this is ready for a pair of eyes :)

**Comment by grantzvolsky** [comment]
This is ready to be reviewed now :)

File: pipeline/authn/authenticator_cookie_session_test.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func TestAuthenticatorCookieSession(t *testing.T) {
 		t.Run("description=should pass through method, path, and headers to auth server", func(t *testing.T) {
 			testServer, requestRecorder := makeServer(200, `{"subject": "123"}`)
 			err := pipelineAuthenticator.Authenticate(
-				makeRequest("PUT", "/users/123?query=string", map[string]string{"sessionid": "zyx"}, ""),
+				makeRequest("PUT", "/users/123", "query=string", map[string]string{"sessionid": "zyx"}, ""),
 				session,
 				json.RawMessage(fmt.Sprintf(`{"check_session_url": "%s"}`, testServer.URL)),
```
Comment (aeneasr) [inline pipeline/authn/authenticator_cookie_session_test.go:63]: Could you please add a test which uses a `check_session_url` that has a query parameter and observes the behaviour dependent on preserve_query?

**Review by aeneasr — CHANGES_REQUESTED** [review]
Looking great! I think we should add test cases which ensures that query parameters from `check_session_url` are indeed preserved (not just empty but actually added) when `preserve_query` is set.

Otherwise, this looks great!

File: pipeline/authn/authenticator_cookie_session_test.go (hunk 1 above)
Comment (grantzvolsky) [reply pipeline/authn/authenticator_cookie_session_test.go:63]: True! It should be covered now :)

**Review by aeneasr — APPROVED** [review]
Awesome, thank you! 🎉
